core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -268,6 +270,31 @@ object UnitAutomation {
         return true
     }
 
+    private fun tryHeadTowardsSiegedCity(unit: MapUnit): Boolean {
+        val siegedCities = unit.civInfo.cities
+                .filter { unit.civInfo == it.civInfo }
+                .filter { it.health < (it.getMaxHealth() * 0.75) } //Weird health issues and making sure that not all forces move to good defenses
+                .filterNot { it.isInResistance() } //Don't stop pushing keep the battle tempo up
+                .asSequence()
+
+        val strongestForce = siegedCities
+                .asSequence().map { it.getCenterTile().getTilesAtDistance(2) }
+                .flatMap { it.asSequence() }
+                .sortedByDescending { it.getInfluence() }
+                .firstOrNull { unit.movement.canReach(it) }
+        
+        if (strongestForce != null) {
+            return headTowardsSiegedCity(unit, strongestForce)
+        }
+        return false
+    }
+    
+    private fun headTowardsSiegedCity(unit: MapUnit, strongestForce: TileInfo): Boolean {
```

Comment:
Not sure what the point of this is

Reply:
Got rid of this. I was copying headTowardsEnemyCity but realized I didn't need the function